extension/src/state_aggregate.rs

```diff
@@ -337,7 +337,7 @@ pub mod toolkit_experimental {
                     state: self.durations.as_slice()[self.first_state as usize]
                         .state
                         .materialize(&states),
-                    time: self.first_time,
+                    time: interval_start,
```

Comment:
This change is needed to prevent the newly added start and range assertions from failing with `interpolated_duration_in`, but I'm not entirely sure that this is actually correct.

Reply:
The failure was from the other assert you commented on:
https://github.com/timescale/timescaledb-toolkit/blob/2db5a02b5d6cc4023ca69858f309d9fb037b3455/extension/src/state_aggregate.rs#L756-L760

I removed the assert and undid this change here.